src/ApparentHorizons/ComputeExcisionBoundaryVolumeQuantities.tpp

```diff
@@ -0,0 +1,206 @@
+          tmpl::list<>>,
+      "Found a dest tag that is not allowed");
+  static_assert(
+      std::is_same_v<
+          tmpl::list_difference<required_dest_tags<TargetFrame>, DestTagList>,
+          tmpl::list<>>,
+      "A required dest tag is missing");
+
+  const auto& psi =
+      get<gr::Tags::SpacetimeMetric<3, Frame::Inertial>>(src_vars);
+
+  if (target_vars->number_of_grid_points() !=
+      src_vars.number_of_grid_points()) {
+    target_vars->initialize(src_vars.number_of_grid_points());
+  }
+
+  using metric_tag = gr::Tags::SpatialMetric<3, TargetFrame>;
+  using inv_metric_tag = gr::Tags::InverseSpatialMetric<3, TargetFrame>;
+  using lapse_tag = gr::Tags::Lapse<DataVector>;
+  using inertial_shift_tag = gr::Tags::Shift<3, Frame::Inertial>;
+
+  // Additional temporary tags used for multiple frames
+  using inertial_metric_tag = gr::Tags::SpatialMetric<3, Frame::Inertial>;
+  using inertial_inv_metric_tag =
+      gr::Tags::InverseSpatialMetric<3, Frame::Inertial>;
+
+  // All of the temporary tags, including some that may be repeated
+  // in the target_variables (for now).
+  using full_temp_tags_list =
+      tmpl::list<metric_tag, inv_metric_tag, lapse_tag, inertial_shift_tag,
+                 inertial_metric_tag,
+                 inertial_inv_metric_tag>;
+
+  // temp tags without variables that are already in DestTagList.
+  using temp_tags_list =
+      tmpl::list_difference<full_temp_tags_list, DestTagList>;
+  TempBuffer<temp_tags_list> buffer(get<0, 0>(psi).size());
+
+  // These may or may not be temporaries, depending on if they are asked for
+  // in target_vars.
+  auto& lapse = *(get<lapse_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_shift = *(get<inertial_shift_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_metric = *(get<inertial_metric_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_inv_metric =
+      *(get<inertial_inv_metric_tag>(
+          target_vars, make_not_null(&buffer)));
+  auto& metric = *(get<metric_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inv_metric = *(get<inv_metric_tag>(
+      target_vars, make_not_null(&buffer)));
+
+  // Actual computation starts here
+
+  gr::spatial_metric(make_not_null(&inertial_metric), psi);
+  // put determinant of 3-metric temporarily into lapse to save memory.
+  determinant_and_inverse(make_not_null(&lapse),
+                          make_not_null(&inertial_inv_metric), inertial_metric);
+
+  // Transform spatial metric
+  transform::to_different_frame(make_not_null(&metric), inertial_metric,
+                                jacobian);
+
+  // Invert transformed 3-metric.
+  // put determinant of 3-metric temporarily into lapse to save memory.
+  determinant_and_inverse(make_not_null(&lapse), make_not_null(&inv_metric),
+                          metric);
+
+  gr::shift(make_not_null(&inertial_shift), psi, inertial_inv_metric);
```

Comment:
don't we want to transform the shift into the `TargetFrame` or are we delaying that change? I'm fine leaving it for a followup PR, but then please just add a comment here.